packages/transaction-controller/CHANGELOG.md

```diff
@@ -7,6 +7,32 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [23.0.0]
+
+### Changed
+
+- **BREAKING:** `approveTransactionsWithSameNonce()` now requires `chainId` to be populated in for each TransactionParams that is passed ([#3643](https://github.com/MetaMask/core/pull/3643))
+- `addTransaction()` now accepts optional `networkClientId` in its options param which specifies the network client that the transaction will be processed with during its lifecycle if the `isMultichainEnabled` feature flag is on ([#3643](https://github.com/MetaMask/core/pull/3643))
+- `estimateGas()` now accepts optional networkClientId as its last param which specifies the network client that should be used to estimate the required gas for the given transaction ([#3643](https://github.com/MetaMask/core/pull/3643))
+- `estimateGasBuffered()` now accepts optional networkClientId as its last param which specifies the network client that should be used to estimate the required gas plus buffer for the given transaction ([#3643](https://github.com/MetaMask/core/pull/3643))
+- `getNonceLock()` now accepts optional networkClientId as its last param which specifies which the network client's nonceTracker should be used to determine the next nonce. ([#3643](https://github.com/MetaMask/core/pull/3643))
+  - When used with the `isMultichainEnabled` feature flag on and with networkClientId specified, this method will also restrict acquiring the next nonce by chainId, i.e. if this method is called with two different networkClientIds on the same chainId, only the first call will return immediately with a lock from its respective nonceTracker with the second call being blocked until the first caller releases its lock
+- `EtherscanRemoteTransactionSource` now enforces a 5 second delay between requests to avoid rate limiting ([#3643](https://github.com/MetaMask/core/pull/3643))
+- `TransactionMeta` type now specifies an optional `networkClientId` field ([#3643](https://github.com/MetaMask/core/pull/3643))
+- `startIncomingTransactionPolling()` now accepts an optional array of `networkClientIds`. When provided, the `IncomingTransactionHelpers` for those networkClientIds will be started. If empty or not provided, the global `IncomingTransactionHelper` will be started. ([#3643](https://github.com/MetaMask/core/pull/3643))
```

Comment:
IncomingTransactionHelper isn't accessible outside of the class, so consumers probably wouldn't know about it. Perhaps it would be better to explain the behavior more generally here?

Reply:
behavior more generally explained with less private implementation detail 

https://github.com/MetaMask/core/pull/3925/commits/b333b53be39f803b160042699fa270894339442e

Reply:
Great!